controllers/humioparser_controller.go

```diff
@@ -81,7 +82,7 @@ func (r *HumioParserReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		if err != nil {
 			return reconcile.Result{}, r.logErrorAndReturn(err, "unable to set cluster state")
 		}
-		return reconcile.Result{}, err
+		return reconcile.Result{RequeueAfter: time.Second * 15}, nil
```

Comment:
@jswoods - does this just retry 1 time and then exit?

Reply:
It will retry every 15 seconds until it works.

Reply:
thanks!

Reply:
@someword Here's a bit of context for how this works:
- If `Reconcile()` returns an error, it gets logged out and it gets requeued immediately.
- If no error is returned, but you still want to requeue immediately you can return `reconcile.Result{Requeue: true}`
- If no error is returned, but you want to requeue after some specified amount of time has passed, you can return `reconcile.Result{RequeueAfter: time.Second * 15}`. Waiting a bit can be helpful in certain cases where there might be a slight delay/race between certain resources becoming available.

There's no knowledge about "previous" runs, so it will just wait 15 seconds and try again no matter how many times the same thing has failed in the past.

Reply:
If you're curious about where it happens: https://github.com/kubernetes-sigs/controller-runtime/blob/985e819db7af1cd17782ca7ad27644d0ca655054/pkg/internal/controller/controller.go#L296-L322

Notice how the "error" case and the "requeue immediately" both uses `c.Queue.AddRateLimited(req)`, whereas "requeue after X time" is different.